front/pages/api/w/[wId]/assistant/agent_configurations/index.ts

```diff
@@ -118,26 +119,25 @@ async function handler(
         sort,
       });
       if (withUsage === "true") {
-        agentConfigurations = await safeRedisClient(async (redis) => {
-          return Promise.all(
-            agentConfigurations.map(
-              async (
-                agentConfiguration
-              ): Promise<LightAgentConfigurationType> => {
-                return {
-                  ...agentConfiguration,
-                  usage: await getAgentUsage(auth, {
-                    providedRedis: redis,
-                    agentConfiguration,
-                    workspaceId: owner.sId,
-                  }),
-                };
-              }
-            )
-          );
+        const mentionCounts = await safeRedisClient(async (redis) => {
+          return getAgentsUsage({
+            providedRedis: redis,
+            workspaceId: owner.sId,
+          });
+        });
+        const usageMap = new Map<string, mentionCount>();
+        mentionCounts.forEach((mentionCount) => {
+          usageMap.set(mentionCount.agentId, mentionCount);
         });
+        agentConfigurations = agentConfigurations.map((agentConfiguration) => ({
+          ...agentConfiguration,
+          usage: {
+            messageCount: usageMap.get(agentConfiguration.sId)?.count || 0,
```

Comment:
nit: AFAICT we don't use `get(`, we access directly `usageMap[configuration.sId]`